sdk/storage/Azure.Storage.Common/src/Shared/ClientsideEncryption/Models/EncryptionData.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Cryptography;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.Cryptography.Models
+{
+    /// <summary>
+    /// Represents the encryption data that is stored on the service.
+    /// </summary>
+    internal class EncryptionData
+    {
+        /// <summary>
+        /// The blob encryption mode.
+        /// </summary>
+        public string EncryptionMode { get; set; }
+
+        /// <summary>
+        /// A <see cref="WrappedKey"/> object that stores the wrapping algorithm, key identifier and the encrypted key.
+        /// </summary>
+        public WrappedKey WrappedContentKey { get; set; }
+
+        /// <summary>
+        /// The encryption agent.
+        /// </summary>
+        public EncryptionAgent EncryptionAgent { get; set; }
+
+        /// <summary>
+        /// The content encryption IV.
+        /// </summary>
+        public byte[] ContentEncryptionIV { get; set; }
+
+#pragma warning disable CA2227 // Collection properties should be read only
+        /// <summary>
+        /// Metadata for encryption. Currently used only for storing the encryption library, but may contain other data.
+        /// </summary>
+        public Metadata KeyWrappingMetadata { get; set; }
+#pragma warning restore CA2227 // Collection properties should be read only
+
+        /// <summary>
+        /// Serializes this object to JSON.
+        /// </summary>
+        /// <returns></returns>
+        public string Serialize()
+            => EncryptionDataSerializer.Serialize(this);
+
+        /// <summary>
+        /// Deserializes an <see cref="EncryptionData"/> from JSON.
+        /// </summary>
+        /// <param name="json">JSON to deserialize.</param>
+        /// <returns></returns>
+        public static EncryptionData Deserialize(string json)
+            => EncryptionDataSerializer.Deserialize(json);
```

Comment:
To match the `Serialize()` instance method. My understanding is that having an instance serialize method and static deserialize method is a good pattern to have on an object intended to be serialized. Should I not be doing this?

Reply:
I usually avoid adding serialize and deserialize method into POCO. The reason is that there might be many serialization strategies at some point (be open for extension). 
So I'd leave the `EncryptionData` as pure model class and have serializer/deserializer as external service/extension.

Reply:
Github needs a reaction or a flag to say "addressed in upcoming push," because I don't want to mark it as resolved when I haven't pushed it or in case it's not actually what the commenter wanted, but continually typing this is annoying.

BTW, addressed in upcoming push.